src/main/java/nl/tudelft/contextproject/Main.java

```diff
@@ -1,161 +1,184 @@
-		for (Iterator<Entity> i = level.getEntities().iterator(); i.hasNext();) {
-			Entity e = i.next();
-		    EntityState state = e.getState();
-			switch (state) {
-			case DEAD:
-				rootNode.detachChild(e.getGeometry());
-				i.remove();
-				continue;
-			case NEW:
-				rootNode.attachChild(e.getGeometry());
-				e.setState(EntityState.ALIVE);
-				e.simpleUpdate(tpf);
-				break;
-			default:
-				e.simpleUpdate(tpf);
-				break;
-			}
-		}
-	}
-
-	/**
-	 * Return the singleton instance of the game.
-	 * @return the running instance of the game.
-	 */
-	public static Main getInstance() {
-		if (instance == null) instance = new Main();
-		return instance;
-	}
-	
-	/**
-	 * Getter for the current level.
-	 * @return The current level.
-	 */
-	public Level getLevel() {
-		return level;
-	}
+package nl.tudelft.contextproject;
+
+import java.util.Iterator;
+
+import com.jme3.app.SimpleApplication;
+import com.jme3.light.Light;
+import com.jme3.scene.Geometry;
+import com.jme3.scene.Node;
+
+import nl.tudelft.contextproject.audio.AudioManager;
+import nl.tudelft.contextproject.audio.BackgroundMusic;
+import nl.tudelft.contextproject.level.DrawableFilter;
+import nl.tudelft.contextproject.level.Level;
+import nl.tudelft.contextproject.level.LevelFactory;
+import nl.tudelft.contextproject.level.MapBuilder;
+import nl.tudelft.contextproject.level.RandomLevelFactory;
+
+/**
+ * Main class of the game 'The Cave of Caerbannog'.
+ */
+public class Main extends SimpleApplication {
+	private static Main instance;
+	private Level level;
+	private LevelFactory levelFactory;
+
+	/**
+	 * Method used for testing.
+	 * Sets the instance of this singleton to the provided instance.
+	 * @param main The new value of the instance.
+	 */
+	static void setInstance(Main main) {
+		instance = main;
+	}
+	
+	/**
+	 * Method used for testing.
+	 * Sets the rootNode of Main to a new Node.
+	 * @param rn The new node to replace the rootNode.
+	 */
+	void setRootNode(Node rn) {
+		rootNode = rn;
+	}
+
+	/**
+	 * Main method that is called when the program is started.
+	 * @param args run-specific arguments.
+	 */
+	public static void main(String[] args) {
+		getInstance().start();
+	}
+
+	@Override
+	public void simpleInitApp() {
+		levelFactory = new RandomLevelFactory(10, 10);
+		setLevel(levelFactory.generateRandom());
+		attachLevel();
+		
+		/* Temp code*/
+		MapBuilder.setLevel(level);
+		DrawableFilter filter = new DrawableFilter(false);
+		filter.addEntity(level.getPlayer());
+		filter.addEntity(new Entity() {
+			@Override
+			public Geometry getGeometry() {
+				 return null;
+			}
+			@Override
+			public void simpleUpdate(float tpf) { }
+
+			@Override
+			public void setGeometry(Geometry geometry) { }
+		});
+		MapBuilder.export("hello.png", filter, 16);
+		
+		//Initialize the AudioManager.
+		AudioManager.getInstance().init();
+		
+		//Start the background music
+		BackgroundMusic.getInstance().start();
+	}
+	
+	@Override
+	public void stop() {
```

Comment:
@BCrlrd Well, the method overrides the stop method from jMonkey's SimpleApplication, so it cannot be renamed. It simply stops the background music before stopping the rest of the game.